src/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredPart.cs

```diff
@@ -226,5 +226,44 @@ public bool TryCloseGenericPart(Type[] typeArguments, out DiscoveredPart closed)
         }
 
         public IEnumerable<DiscoveredExport> DiscoveredExports { get { return _exports; } }
+
+        // uses the fact that current usage only has comparisons
+        // between ParameterInfo objects from the same constructor reference,
+        // thus only the position needs to be compared.
+        // Equals checks the member reference equality in case usage changes.
+        private sealed class ParameterInfoComparer : IEqualityComparer<ParameterInfo>
+        {
+            public static readonly ParameterInfoComparer Instance = new ParameterInfoComparer();
+
+            public int GetHashCode(ParameterInfo obj)
+            {
+                return obj.Position.GetHashCode();
```

Comment:
My preference would be to not rely on the current usage of this when designing the comparer.